runtime/src/enclave_rpc/client.rs

```diff
@@ -109,8 +109,16 @@ impl Controller {
                 Command::Call(request, kind, sender, retries) => {
                     self.call(request, kind, sender, retries).await
                 }
-                Command::PeerFeedback(pfid, peer_feedback) => {
+                Command::PeerFeedback(pfid, peer_feedback, kind) => {
                     self.transport.set_peer_feedback(pfid, Some(peer_feedback));
+
+                    // In case the peer feedback is bad, reset the session so a new peer can be
+                    // selected for a subsequent session.
+                    if !matches!(peer_feedback, types::PeerFeedback::Success)
```

Comment:
Feels like this doesn't belong here, if we even need this.

Reply:
How will bad peer feedback from the application layer be taken into account then? As currently the node will be fixed unless a transport error happens.

Reply:
You are right, this is probably the way to go. However, I'm not sure if all errors on application layer should disconnect the underlying node. as connect calls are expensive (don't have an example for that, except when we replicate secrets, but there we switch to the next node if the current node returns an error). Should we let the caller decide that?

Reply:
Yeah the caller should decide, same as in the Go version.

Reply:
Added a way for the caller to provide feedback, please take a look.